geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CompactDiskStoreCommand.java

```diff
@@ -0,0 +1,185 @@
+
+        Map<DistributedMember, PersistentID> overallCompactInfo = new HashMap<>();
+
+        Set<?> otherMembers = ds.getDistributionManager().getOtherNormalDistributionManagerIds();
+        Set<InternalDistributedMember> allMembers = new HashSet<>();
+
+        for (Object member : otherMembers) {
+          allMembers.add((InternalDistributedMember) member);
+        }
+        allMembers.add(ds.getDistributedMember());
+
+        String groupInfo = "";
+        // if groups are specified, find members in the specified group
+        if (groups != null && groups.length > 0) {
+          groupInfo = CliStrings.format(CliStrings.COMPACT_DISK_STORE__MSG__FOR_GROUP,
+              new Object[] {Arrays.toString(groups) + "."});
+          final Set<InternalDistributedMember> selectedMembers = new HashSet<>();
+          List<String> targetedGroups = Arrays.asList(groups);
+          for (InternalDistributedMember member : allMembers) {
+            List<String> memberGroups = member.getGroups();
+            if (!Collections.disjoint(targetedGroups, memberGroups)) {
+              selectedMembers.add(member);
+            }
+          }
+
+          allMembers = selectedMembers;
+        }
+
+        // allMembers should not be empty when groups are not specified - it'll
+        // have at least one member
+        if (allMembers.isEmpty()) {
+          result = ResultBuilder.createUserErrorResult(
+              CliStrings.format(CliStrings.COMPACT_DISK_STORE__NO_MEMBERS_FOUND_IN_SPECIFED_GROUP,
+                  new Object[] {Arrays.toString(groups)}));
+        } else {
+          // first invoke on local member if it exists in the targeted set
+          if (allMembers.remove(ds.getDistributedMember())) {
+            PersistentID compactedDiskStoreId = CompactRequest.compactDiskStore(diskStoreName);
+            if (compactedDiskStoreId != null) {
+              overallCompactInfo.put(ds.getDistributedMember(), compactedDiskStoreId);
+            }
+          }
+
+          // was this local member the only one? Then don't try to send
+          // CompactRequest. Otherwise, send the request to others
+          if (!allMembers.isEmpty()) {
+            // Invoke compact on all 'other' members
+            Map<DistributedMember, PersistentID> memberCompactInfo =
+                CompactRequest.send(ds.getDistributionManager(), diskStoreName, allMembers);
+            if (memberCompactInfo != null && !memberCompactInfo.isEmpty()) {
+              overallCompactInfo.putAll(memberCompactInfo);
+              memberCompactInfo.clear();
+            }
+            String notExecutedMembers = CompactRequest.getNotExecutedMembers();
+            if (notExecutedMembers != null && !notExecutedMembers.isEmpty()) {
+              LogWrapper.getInstance()
+                  .info("compact disk-store \"" + diskStoreName
+                      + "\" message was scheduled to be sent to but was not send to "
```

Comment:
I realize this goes beyond a simple class-extraction refactor, this message is poorly worded. A couple alternatives:

"message was scheduled to be sent but was not for "
   or
"message was scheduled to be sent to but was not sent to "